tools/topology/m4/mux.m4

```diff
@@ -0,0 +1,49 @@
+divert(-1)
+
+dnl Define macro for mux widget
+
+dnl Mux name)
+define(`N_MUX', `MUX'PIPELINE_ID`.'$1)
+
+dnl W_MUX(name, format, periods_sink, periods_source, kcontrol_list)
+define(`W_MUX',
+`SectionVendorTuples."'N_MUX($1)`_tuples_w" {'
+`	tokens "sof_comp_tokens"'
+`	tuples."word" {'
+`		SOF_TKN_COMP_PERIOD_SINK_COUNT'		STR($3)
+`		SOF_TKN_COMP_PERIOD_SOURCE_COUNT'	STR($4)
+`	}'
+`}'
+`SectionData."'N_MUX($1)`_data_w" {'
+`	tuples "'N_MUX($1)`_tuples_w"'
+`}'
+`SectionVendorTuples."'N_MUX($1)`_tuples_str" {'
+`	tokens "sof_comp_tokens"'
+`	tuples."string" {'
+`		SOF_TKN_COMP_FORMAT'	STR($2)
+`	}'
+`}'
+`SectionVendorTuples."'N_MUX($1)`_mux_process_tuples_str" {'
+`	tokens "sof_process_tokens"'
+`	tuples."string" {'
+`		SOF_TKN_PROCESS_TYPE'	"MUX"
+`	}'
+`}'
+`SectionData."'N_MUX($1)`_data_str" {'
+`	tuples "'N_MUX($1)`_tuples_str"'
+`	tuples "'N_MUX($1)`_mux_process_tuples_str"'
+`}'
+`SectionWidget."'N_MUX($1)`" {'
+`	index "'PIPELINE_ID`"'
+`	type "effect"'
+`	no_pm "true"'
+`	data ['
+`		"'N_MUX($1)`_data_w"'
+`		"'N_MUX($1)`_data_str"'
+`	]'
+`	bytes ['
+		$5
+`	]'
+`}')
+
+divert(0)dnl
```

Comment:
@juimonen can we have just one mux.m4 with the token for mux or demux passed as a parameter?

Reply:
@ranj063 yes sure, it was in my mind, but was hacking this quickly to get something done and had to modify from @akloniex mux to demux.